src/gwt/src/org/rstudio/studio/client/server/remote/ClientEventDispatcher.java

```diff
@@ -214,11 +214,13 @@ else if (type.equals(ClientEvent.ConsoleOutput))
          {
             ConsoleText output = event.getData();
             eventBus_.fireEvent(new ConsoleWriteOutputEvent(output));
+            eventBus_.fireEventToAllSatellites(new ConsoleWriteOutputEvent(output));
          }
          else if (type.equals(ClientEvent.ConsoleError))
          {
             ConsoleText error = event.getData();
             eventBus_.fireEvent(new ConsoleWriteErrorEvent(error));
+            eventBus_.fireEventToAllSatellites(new ConsoleWriteErrorEvent(error));
```

Comment:
The issue here was that events where being forwarded to the satellite window through two different queues and the events where arriving in different order. Specifically, `setChunkExecuting` was not arriving first and this call has the side-effect of initializing console event listeners in the chunk. I moved away from forwarding the chunk outputs directly into the window and instead, are using now the client event dispatcher which fixes this issue by maintaining order.